File: java-api-core/src/com/kinvey/java/sync/dto/SyncItem.java
package com.kinvey.java.sync.dto;

import com.google.api.client.json.GenericJson;
import com.google.api.client.util.Key;

import java.io.IOException;

/**
 * Created by yuliya on 10/04/17.
 */

public class SyncItem extends SyncRequest{

    @Key("requestMethod")
    private String requestMethod;

    public SyncItem() {
    }

    public SyncItem(SyncItem.HttpVerb httpMethod, SyncMetaData entityID, String collectionName) {
        this.requestMethod = httpMethod.name();
        this.id = entityID;
        this.collectionName = collectionName;
    }

    public SyncItem.HttpVerb getRequestMethod() {
        return SyncItem.HttpVerb.fromString(requestMethod);
    }

    public void setRequestMethod(SyncItem.HttpVerb requestMethod) {
        this.requestMethod = requestMethod.name();
    }

    public GenericJson getEntity() {
        GenericJson entity = null;
        SyncMetaData entityId = getEntityID();
        if (entityId != null) {
            entity = entityId.getEntity();
        }
        return entity;
    }
}

File: java-api-core/src/com/kinvey/java/sync/dto/SyncItem.kt
package com.kinvey.java.sync.dto

import com.google.api.client.json.GenericJson
import com.google.api.client.util.Key

/**
 * Created by yuliya on 10/04/17.
 */

open class SyncItem : SyncRequest {
    @Key("requestMethod")
    private var requestMethodStr: String? = null

    constructor() {}
    constructor(httpMethod: HttpVerb?, entityID: SyncMetaData, collectionName: String?) {
        requestMethod = httpMethod
        this.entityID = entityID
        this.collectionName = collectionName
    }

    var requestMethod: HttpVerb? = null
        get() {
            return HttpVerb.fromString(requestMethodStr)
        }
        set (value) {
            field = value
            requestMethodStr = value?.name
        }

    val entity: GenericJson?
        get() {
            var entity: GenericJson? = null
            val entityId = entityID
            if (entityId != null) {
                entity = entityId.entity
            }
            return entity
        }
}